py/torch_tensorrt/dynamo/backend/test/utils.py

```diff
@@ -31,6 +34,8 @@ def fx_dynamo_testing_backend(
         torch_executed_ops=torch_executed_ops,
     )
 
+    gm = pre_aot_module_replacement(gm)
```

Comment:
Why is there a separate testing backend? Would we need to continue to make changes to this in step with the actual one?

Reply:
The main reason for the separate testing backend is the argument `store_intermediate_graphs`. This is needed to track intermediate partitioned modules at different points in the compilation, to ensure decompositions, fusions, etc. are being utilized.

As changes are made to the main backend, yes, those changes would need to be reflected here, and in the `compile_module_testing` function below, however these are higher-level functions which do not change often.